### PR TITLE
Add protection for metachain over computation

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Added protection for metachain when it is stuck because of some high load computation
* Fixed panic situation in indexHashedNodesCoordinator when nodesConfig is nil
* Fixed situation when cross mini blocks and cross transactions from shards to meta were not broadcast by proposer at commit time, but only set for delayed broadcast

### CODE REVIEW

File: consensus/spos/bls/subroundEndRound.go [hunk 1]
```diff
@@ -376,15 +377,66 @@ func (sr *subroundEndRound) broadcastMiniBlocksAndTransactions() error {
...
+		return sr.broadcast(metaMiniBlocks, metaTransactions)
 	}
 
-	err = sr.BroadcastMessenger().BroadcastMiniBlocks(miniBlocks)
-	if err != nil {
-		return err
+	return sr.broadcast(miniBlocks, transactions)
+}
+
+func (sr *subroundEndRound) extractMetaMiniBlocksAndTransactions(
```
Comment (sasurobert) [inline consensus/spos/bls/subroundEndRound.go:393]: this became a super special think. might rethink sometime a refactor of consensus, to be more general.

Comment (sasurobert) [reply consensus/spos/bls/subroundEndRound.go:393]: Maybe add a unit test :D

Comment (SebastianMarian) [reply consensus/spos/bls/subroundEndRound.go:393]: Actually the caller method already do things which depend of shardCoordinator.SelfId(), as we wanted to keep only one implementation for all shards

Comment (iulianpascalau) [reply consensus/spos/bls/subroundEndRound.go:393]: Could have become more general by providing the shard ID. For now it will suffice.

**Pull request merged**